lib/blkfront.ml

```diff
@@ -325,9 +325,20 @@ let resume () =
     resume v
   ) devs
 
-let disconnect _id =
-  Log.err (fun f -> f "Blkfront: disconnect not implement yet");
-  return ()
+let disconnect t =
+  (* when we write here, it ends up in the dom0 tree, not our local one *)
+  (* that's seeming mirrored from the actual write location, local/domid/device/vbd/blah *)
+  (* do we need to write it somewhere else?  34/device/vbd/blah shows *)
+  (* the second set of things is in the libxl/ tree - how can we manipulate that? *)
```

Comment:
I'm not sure how this is supposed to work either. It seems a bit pointless to write the `Closing` state and then delete the whole tree immediately (probably before the backend sees it). It looks like mini-os does this:

1. Set frontend state to `Closing`.
2. Wait for backend state to become `>= Closing`.
3. Set frontend state to `Closed`.
4. Wait for backend state to become `>= Closed`.
5. Set frontend state to `Initialising`.
6. Wait for backend state to become `>= InitWait && < Closed`.

(https://github.com/mirage/mini-os/blob/master/blkfront.c#L247)

Reply:
Ha, oops, that's some comment garbage I didn't intend to submit with this PR.  Anyway, thanks for the pointer on the flow here - I consulted some Xen experts and got some subset of that workflow, then did a little hacking around and submitted what seems to work.  What you describe does indeed seem more sensible.